### ISSUE TITLE
[Types] @types/react 18.0.0 causes children type issue in AnimatePresence

### ISSUE THREAD

**Issue by IljaDaderko** [nandorojo/moti#187]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

I believe there were some changes made to `children` type in the new react types package that causes following typescript error in `AnimatePresence` component (I think it needs to explicitly define children now?)

```
Type '{ children: false | Element | undefined; }' has no properties in common with type 'IntrinsicAttributes & AnimatePresenceProps'.
```

### Expected Behavior

Should not error when children are passed

### Steps To Reproduce

1. Import `AnimatePresence` component
2. Add some children to it
3. Update to latest `@types/react` package

### Versions

```markdown
- Moti: 0.17.1
- Reanimated: 2.6.0
- React Native: 0.67.4
```

### Screenshots

<img width="733" alt="Screenshot 2022-04-09 at 10 44 48" src="https://user-images.githubusercontent.com/3154053/162562167-8e282ce0-c08f-4071-9f62-ca26bb214c51.png">

### Reproduction

-

**Comment by nandorojo** [comment]
this is a framer motion issue. i might have to update the dependency

**Comment by nandorojo** [comment]
can you run `yarn why framer-motion` and share the result?

**Comment by IljaDaderko** [comment]
Sure thing, here you go

```zsh
framer-motion@3.10.6
node_modules/framer-motion
  framer-motion@"^3.9.1" from @motify/core@0.17.1
  node_modules/@motify/core
    @motify/core@"^0.17.1" from moti@0.17.1
    node_modules/moti
      moti@"0.17.1" from @skimitar/app@1.0.0
      @app
        @skimitar/app@1.0.0
        node_modules/@skimitar/app
          workspace @app from the root project
    @motify/core@"^0.17.1" from @motify/components@0.17.1
    node_modules/@motify/components
      @motify/components@"^0.17.1" from moti@0.17.1
      node_modules/moti
        moti@"0.17.1" from @skimitar/app@1.0.0
        @app
          @skimitar/app@1.0.0
          node_modules/@skimitar/app
            workspace @app from the root project
```

**Comment by IljaDaderko** [comment]
Here is related post from dan on stackoverflow
https://stackoverflow.com/questions/71788254/react-18-typescript-children-fc/71809927#71809927

**Comment by IljaDaderko** [comment]
Looks like this was fixed upstream if moti updates framer-motion?
https://twitter.com/mattgperry/status/1513933956877627407

**Comment by nandorojo** [comment]
sounds good, until then you can use a yarn resolution for it. i’ll update when i get the chance. thanks for the catch!

**Comment by nandorojo** [comment]
Fixed in `0.18.0-alpha.17`.

`yarn add moti@canary`
